homeassistant/components/myuplink/sensor.py

```diff
@@ -103,3 +155,34 @@ def native_value(self) -> StateType:
         """Sensor state value."""
         device_point = self.coordinator.data.points[self.device_id][self.point_id]
         return device_point.value  # type: ignore[no-any-return]
+
+
+class MyUplinkEnumSensor(MyUplinkDevicePointSensor):
+    """Representation of a myUplink device point sensor for ENUM device_class."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SensorEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            device_point=device_point,
+            entity_description=entity_description,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        self._attr_options = [x["text"].capitalize() for x in device_point.enum_values]
+
+    @property
+    def native_value(self) -> StateType:
+        """Sensor state value for enum sensor."""
+        device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        return {x["value"]: x["text"] for x in device_point.enum_values}[  # type: ignore[no-any-return]
```

Comment:
If the device point enum values are static after entity creation, we could consider storing the dict that maps value to text in an attribute on the entity, so we don't need to create that every state update.

Reply:
It is reasonable to believe that enum_values are static between HA restarts. But there is really no way to know for sure. The MyUplink help desk explicitly says that they do not provide any support for the public API. They just point me at the Swagger site.
But that said, I think it is reasonable to do as you suggest.